### PR TITLE
Add pytest tests and  a GitHub Actions workflow for SD register decoding

### PR DESCRIPTION
**Pull request by jluebbe**
This builds on #62.

### CODE REVIEW

File: tests/reference/1b534d474638533530d8466363a16700.json [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+    "fields": [
+      {
+        "field": "MID",
+        "name": "Manufacturer ID",
+        "raw": [
+          27,
+          8
+        ],
+        "enum": null
+      },
```
Comment (hnez) [inline tests/reference/1b534d474638533530d8466363a16700.json:111]: Do you know the manufacturer of this card? Or is it one of the no-name unbranded varieties?

File: .github/workflows/pytest.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+jobs:
+  python-pytest:
+    name: Python pytest
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install python dependencies
+        run: |
+          python -m pip install --upgrade pip virtualenv
+          python -m pip install --upgrade pytest
```
Comment (hnez) [inline .github/workflows/pytest.yml:14]: We do not use `virtualenv`, do we?
And `python3-pytest` is in debian. Couldn't we just `sudo apt install python3-pytest`?
(I guess the same comment could be made if we did `apt install` here. "Could'nt we just pip install instead?". Maybe it just comes down to personal preference)

**Review by hnez — COMMENTED** [review]
Very nice.

This means https://github.com/linux-automation/usbsdmux/pull/62#issuecomment-1860456511 is done.
Once #62 is merged this has my approval.

File: tests/reference/1b534d474638533530d8466363a16700.json (hunk 1 above)
Comment (jluebbe) [reply tests/reference/1b534d474638533530d8466363a16700.json:111]: It's a Samsung, which is not known to the decoder yet.

File: .github/workflows/pytest.yml (hunk 2 above)
Comment (jluebbe) [reply .github/workflows/pytest.yml:14]: Yes, virtualenv is not used. I'd prefer to install from pypi, as it's easier to get new versions that way.

**Review by SmithChart — APPROVED** [review]
Looks good! Thanks for adding tests for decoding!

**Comment by jluebbe** [comment]
Rebased on #62.

**Comment by SmithChart** [comment]
@jluebbe Please rebase

**Comment by jluebbe** [comment]
Rebased.

**Review by SmithChart — APPROVED** [review]

**Pull request merged**